packages/lit-dev-content/samples/examples/react-basics/demo-greeting.ts

```diff
@@ -8,7 +8,32 @@ export class DemoGreeting extends LitElement {
   render() {
     return html`
       <p>Hello, ${this.name}!</p>
-      <p>Checkout our <a href="https://lit.dev/docs/frameworks/react/" target="_blank" rel="noopener noreferrer">React framework docs</a>!</p>
+      <p>
+        Checkout our
+        <a
+          href="https://lit.dev/docs/frameworks/react/"
+          target="_blank"
+          rel="noopener noreferrer"
+          >React framework docs</a
+        >!
+      </p>
     `;
   }
 }
+
+declare global {
+  interface HTMLElementTagNameMap {
+    'demo-greeting': DemoGreeting;
+  }
+
+  namespace JSX {
```

Comment:
Worth adding a comment that this is for good TypeScript types in JSX? (This is my assumption for what this does).